aten/src/ATen/native/miopen/Conv.cpp

```diff
@@ -0,0 +1,1209 @@
+    /*miopenConvolutionFwdPreference_t pref = cudnn_convolution_FWD_PREFER_FASTEST;
+    MIOPEN_CHECK(miopenGetConvolutionForwardAlgorithm(
+        args.handle,
+        args.idesc.desc(),
+        args.wdesc.desc(),
+        args.cdesc.desc(),
+        args.odesc.desc(),
+        pref,
+        0,
+        algo));*/
+  }
+
+  static void getWorkspaceSize(
+    const ConvolutionArgs& args,
+    algo_t algo, size_t* workspaceSize)
+  {
+    throw std::runtime_error("miopenGetConvolutionForwardWorkspaceSize is not supported.");
+    /*MIOPEN_CHECK(miopenGetConvolutionForwardWorkspaceSize(
+        args.handle,
+        args.idesc.desc(),
+        args.wdesc.desc(),
+        args.cdesc.desc(),
+        args.odesc.desc(),
+        algo,
+        workspaceSize));*/
+  }
+};
+
+template<>
+struct algorithm_search<miopenConvBwdDataAlgorithm_t> {
+  using perf_t = miopenConvAlgoPerf_t;
+  using algo_t = miopenConvBwdDataAlgorithm_t;
+
+  static constexpr auto DEFAULT_ALGO = miopenConvolutionBwdDataAlgoGEMM;
+  static BenchmarkCache<algo_t>& cache() { return bwd_data_algos; }
+
+  static perf_t findAlgorithm(const ConvolutionArgs& args) {
+    static const algo_t algos[] = {
+        miopenConvolutionBwdDataAlgoGEMM,
+        miopenConvolutionBwdDataAlgoDirect,
+        miopenConvolutionBwdDataAlgoFFT,
+        miopenConvolutionBwdDataAlgoWinograd,
+    };
+    static constexpr int num_algos = 4;
+    static_assert(sizeof(algos) / sizeof(algos[0]) == num_algos,
+                  "Missing MIOpen convolution backward data algorithms.");
+    int perf_count;
+    std::unique_ptr<perf_t[]> perf_results(new perf_t[num_algos]);
+    size_t max_ws_size = getMaxWorkspaceSize(args, algos, num_algos);
+    Workspace ws(max_ws_size);
+    MIOPEN_CHECK(miopenFindConvolutionBackwardDataAlgorithm(
+        args.handle,
+        args.odesc.desc(), args.output.data_ptr(),
+        args.wdesc.desc(), args.weight.data_ptr(),
+        args.cdesc.desc(),
+        args.idesc.desc(), args.input.data_ptr(),
+        num_algos,
+        &perf_count,
+        &perf_results,
+        ws.data,
+        ws.size,
+        false));
+    return getBestAlgorithm(perf_results.get(), args.params.deterministic, perf_count);
+  }
+
+  static void getAlgorithm(const ConvolutionArgs& args, algo_t* algo) {
+    throw std::runtime_error("miopenGetConvolutionBackwardDataAlgorithm is not supported.");
+
+    /*MIOPEN_CHECK(miopenGetConvolutionBackwardDataAlgorithm(
+        args.handle,
+        args.wdesc.desc(),
+        args.odesc.desc(),
+        args.cdesc.desc(),
+        args.idesc.desc(),
+        cudnn_convolution_BWD_DATA_PREFER_FASTEST,
+        0,
+        algo));*/
+  }
+
+  static void getWorkspaceSize(
+    const ConvolutionArgs& args,
+    miopenConvBwdDataAlgorithm_t algo, size_t* workspaceSize)
+  {
+    throw std::runtime_error("miopenGetConvolutionBackwardDataWorkspaceSize is not supported.");
```

Comment:
These workspace functions are all internal ones, so if miopen doesn't need them, get rid of them.